Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -21,981 +21,660 @@
  @Status Stub
  @Notes
 */
-bool CGImageIsMask(CGImageRef image) {
-    UNIMPLEMENTED();
-    return StubReturn();
+const CGFloat *CGImageGetDecode(CGImageRef image) {
+  RETURN_NULL_IF(!image);
+  UNIMPLEMENTED();
+  return StubReturn();
 }
 
 /**
  @Status Stub
  @Notes
 */
-CGImageRef CGImageCreateWithMaskingColors(CGImageRef image, const CGFloat* components) {
-    UNIMPLEMENTED();
-    return StubReturn();
-}
\ No newline at end of file
+CGImageRef CGImageCreateWithMaskingColors(CGImageRef image,
+                                          const CGFloat *components) {
+  RETURN_NULL_IF(!image);
+  UNIMPLEMENTED();
+  return StubReturn();
+}
+
+#pragma region WIC_HELPERS
+
+CGImageRef _CGImageGetImageFromData(void *data, int length) {
+  unsigned char *bytes = static_cast<unsigned char *>(data);
+
+  // TODO: (The below code is not really needed.)
+  if ((bytes[0] == 'I' && bytes[1] == 'I') ||
+      (bytes[0] == 'M' && bytes[1] == 'M')) {
+    return _CGImageLoadTIFF(bytes, length);
+  }
+
+  if (bytes[0] == 'G' && bytes[1] == 'I' && bytes[2] == 'F') {
+    return _CGImageLoadGIF(bytes, length);
+  }
+
+  if (bytes[0] == 'B' && bytes[1] == 'M') {
+    return _CGImageLoadBMP(bytes, length);
+  }
+
+  if ((bytes[0] == 0xFF && bytes[1] == 0xD8) ||
+      (bytes[0] == 0xD8 && bytes[1] == 0xFF)) {
+    return _CGImageLoadJPEG(bytes, length);
+  }
+
+  if (bytes[0] == 0x89 && bytes[1] == 'P' && bytes[2] == 'N' &&
+      bytes[3] == 'G') {
+    return _CGImageLoadPNG(bytes, length);
+  }
+
+  return _CGImageLoadImageWithWICDecoder(GUID_NULL, data, length);
+}
+
+CGImageRef _CGImageLoadGIF(void *bytes, int length) {
+  return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatGif, bytes,
+                                         length);
+}
+
+CGImageRef _CGImageLoadBMP(void *bytes, size_t length) {
+  return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatBmp, bytes,
+                                         length);
+}
+
+CGImageRef _CGImageLoadTIFF(void *bytes, int length) {
+  return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatTiff, bytes,
+                                         length);
+}
+
+CGImageRef _CGImageLoadPNG(void *bytes, int length) {
+  return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatPng, bytes,
+                                         length);
+}
+
+CGImageRef _CGImageLoadJPEG(void *bytes, int length) {
+  return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatJpeg, bytes,
+                                         length);
+}
+
+CGImageRef _CGImageLoadImageWithWICDecoder(REFGUID decoderCls, void *bytes,
+                                           int length) {
+  ComPtr<IWICImagingFactory> imageFactory = _GetWICFactory();
+
+  ComPtr<IWICBitmapDecoder> pDecoder;
+
+  if (decoderCls != GUID_NULL) {
+    ComPtr<IStream> spStream;
+    RETURN_NULL_IF_FAILED(::CreateStreamOnHGlobal(NULL, FALSE, &spStream));
+
+    ULONG written = 0;
+    HRESULT hr;
+
+    hr = spStream->Write(bytes, length, &written);
```

Comment:
looks like the comment is off. It looks fine as is.

---
In reply to: [84144125](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84144125) [](ancestors = 84144125)